packages/yourloops/test/integration/pages/patient-view/trends/trends-view.common.spec.tsx

```diff
@@ -84,11 +91,29 @@ describe('Trends view for anyone', () => {
       // Check Layout
       checkTrendsLayout()
       await checkTrendsBolusAndCarbsAverage()
+    })
 
-      await userEvent.click(screen.getByTestId('button-nav-back'))
+    it('should filter the data on weekday if the user removes one day', async () => {
```

Comment:
The 2 newly added tests could be added to the one above.
For various reason:
   - Run faster (no need to initialize the page many times)
   - Closer to user behaviour/experience (a user makes many actions one after the others, he does not refresh the page between each action)
   - Avoid asserting many times the same things
   - More [here](https://kentcdodds.com/blog/write-fewer-longer-tests)

We have some kind of architecture for this, I can show it to you :slightly_smiling_face:

Reply:
OK